hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableMetaClient.java

```diff
@@ -680,15 +680,12 @@ public void initializeBootstrapDirsIfNotExists() throws IOException {
 
   private static HoodieTableMetaClient newMetaClient(Configuration conf, String basePath, boolean loadActiveTimelineOnLoad,
       ConsistencyGuardConfig consistencyGuardConfig, Option<TimelineLayoutVersion> layoutVersion,
-      String payloadClassName, String recordMergerStrategy, FileSystemRetryConfig fileSystemRetryConfig, Properties props) {
-    HoodieMetaserverConfig metaserverConfig = null == props
-        ? new HoodieMetaserverConfig.Builder().build()
-        : new HoodieMetaserverConfig.Builder().fromProperties(props).build();
+      String payloadClassName, String recordMergerStrategy, FileSystemRetryConfig fileSystemRetryConfig, HoodieMetaserverConfig metaserverConfig) {
     return metaserverConfig.isMetaserverEnabled()
         ? (HoodieTableMetaClient) ReflectionUtils.loadClass("org.apache.hudi.common.table.HoodieTableMetaserverClient",
-        new Class<?>[] {Configuration.class, ConsistencyGuardConfig.class, String.class, FileSystemRetryConfig.class, String.class, String.class, HoodieMetaserverConfig.class},
-        conf, consistencyGuardConfig, recordMergerStrategy, fileSystemRetryConfig,
-        props.getProperty(HoodieTableConfig.DATABASE_NAME.key()), props.getProperty(HoodieTableConfig.NAME.key()), metaserverConfig)
+        new Class<?>[] {Configuration.class, String.class, ConsistencyGuardConfig.class, String.class, FileSystemRetryConfig.class,
+            String.class, String.class, HoodieMetaserverConfig.class}, conf, basePath, consistencyGuardConfig, recordMergerStrategy, fileSystemRetryConfig,
+        metaserverConfig.getDatabaseName(), metaserverConfig.getTableName(), metaserverConfig)
```

Comment:
you already passed `HoodieMetaserverConfig`, then no need to pass database name and table name separately. pls minimize number of args

Reply:
ok this is a separate issue with `HoodieTableMetaserverClient`, we can ignore for now